test/spec/modules/userId_spec.js

```diff
@@ -1511,6 +1517,79 @@ describe('User ID', function () {
       }, {adUnits});
     });
 
+    it('test sharedid via pubcid', function (done) {
+      coreStorage.setCookie('pubcid', 'testpubcid', (new Date(Date.now() + 5000).toUTCString()));
+      coreStorage.setCookie('pubcid_sharedid', 'testsharedid', (new Date(Date.now() + 5000).toUTCString()));
+
+      setSubmoduleRegistry([pubCommonIdSubmodule]);
+      init(config);
+      config.setConfig(getConfigMock(['pubCommonId', 'pubcid', 'cookie']));
+
+      requestBidsHook(function () {
+        adUnits.forEach(unit => {
+          unit.bids.forEach(bid => {
+            // verify that the PubCommonId id data was copied to bid
+            expect(bid).to.have.deep.nested.property('userId.pubcid');
+            expect(bid.userId.pubcid).to.equal('testpubcid');
+            expect(findEid(bid.userIdAsEids, 'pubcid.org')).to.deep.equal(
+              {'source': 'pubcid.org', 'uids': [{'id': 'testpubcid', 'atype': 1}]}
+            );
+            // verify that the sharedid was also copied to bid
+            expect(bid).to.have.deep.nested.property('userId.sharedid');
+            expect(bid.userId.sharedid).to.deep.equal({id: 'testsharedid'});
+            expect(findEid(bid.userIdAsEids, 'sharedid.org')).to.deep.equal(
+              {'source': 'sharedid.org', 'uids': [{'id': 'testsharedid', 'atype': 1}]}
+            );
+          });
+        });
+        coreStorage.setCookie('pubcid', '', EXPIRED_COOKIE_DATE);
+        coreStorage.setCookie('pubcid_sharedid', '', EXPIRED_COOKIE_DATE);
+
+        done();
+      }, {adUnits});
+    });
+
+    it('test pubcid coexisting with sharedid', function (done) {
+      coreStorage.setCookie('pubcid', 'testpubcid', (new Date(Date.now() + 5000).toUTCString()));
+      coreStorage.setCookie('pubcid_sharedid', 'test111', (new Date(Date.now() + 5000).toUTCString()));
+      coreStorage.setCookie('sharedid', JSON.stringify({
+        'id': 'test222',
+        'ts': 1590525289611
+      }), (new Date(Date.now() + 5000).toUTCString()));
+
+      // When both pubcommon and sharedid are configured, pubcommon are invoked first due to
+      // module loading order.  This mean the output from the primary sharedid module will overwrite
```

Comment:
Appears to be alphabetical, and different from the order defined in .submodules.json.  Yeah this bothers me as well.  The two modules receives the same sharedid value from the same endpoint, so even when the order was swapped, the result is pretty much the same.  The only difference is that the main sharedid module adds an extra property called 'third'.  Maybe @jdwieland8282 can comment on whether this is a big deal or not. 

But to be on the safe side, I am thinking of adding a config option to disable sharedid in pubcid module.

Reply:
@pycnvr this is not a big deal, the main sharedid module sets two ids one in the 1p context (ext.eids.uids.id) and one in the 3p context (ext.eids.uids.ext.third).  The latter is the response from sharedid.org/usyc, which can go in the PC module in the more traditional eids blob under sharedid.org

```
{
   "eids":[
      {
         "source":"sharedid.org",
         "uids":[
            {
               "id":"<response from sharedid.org/usyc",
               "atype":1
            }
         ]
      }
   ]
}
```

Reply:
Hi @jdwieland8282, not sure I understand.  😞 Should we add a second ajax call to sharedid.org/usyc?

Reply:
No, I don't think so, I'm just trying to illustrate where the sharedid.org uid should go in the pubcommon module.

Reply:
> But to be on the safe side, I am thinking of adding a config option to disable sharedid in pubcid module. Edit: A separate PR probably.

I think this should definitely be included in this PR...I'd actually argue that the default behavior should be NOT to include sharedid and only via a config should it be added via pubcommon. Pubs may not have the right privacy policies or may have other issues with adding an identity module without their knowledge or direct setup. These changes are adding net new requests to systems that the publisher previously was not aware of.

Reply:
I strongly disagree @smenzer Prebid has its own Privacy policy and pubs can turn this off if they desire.  The purpose of merging the two modules is so that the Prebid community can have access to a fully seeded supported open source identity, that won't happen if the module is off by default.

Reply:
http://docs.prebid.org/privacy.html